File: captainhcg/go/370-range-addition.go
func getModifiedArray(length int, updates [][]int) []int {
    helper := make([]int, length+1)
    for _, arr := range updates{
        st, ed, offset := arr[0], arr[1], arr[2]
        helper[st] += offset
        helper[ed + 1] -= offset
    }
    mod := 0
    for idx, v := range helper{
        mod += v
        helper[idx] = mod
    }
    return helper[:length]
}

File: captainhcg/py/370-range-addition.py
class Solution(object):
    def getModifiedArray(self, length, updates):
        ret = [0] * length
        for st, ed, offset in updates:
            ret[st] += offset
            if ed+1 < length:
                ret[ed+1] -= offset
        mod = 0
        for idx, v in enumerate(ret):
            mod += v
            ret[idx] = mod
        return ret
